allauth/utils.py

```diff
@@ -43,8 +45,13 @@ def _generate_unique_username_base(txts, regex=None):
         username = username.split('@')[0]
         username = username.strip()
         username = re.sub('\s+', '_', username)
-        if username:
-            break
+        # Finally, validating base username without database lookups etc.
+        try:
+            username = adapter.clean_username(username, shallow=True)
+            if username:
```

Comment:
The `if` is not necessary, as `clean_username` always returns a cleaned username, and in case of error, raises an exception.